crates/bevy_asset/src/id.rs

```diff
@@ -178,6 +178,13 @@ impl<A: Asset> From<&Handle<A>> for AssetId<A> {
     }
 }
 
+impl<A: Asset> From<&mut Handle<A>> for AssetId<A> {
```

Comment:
Can we do something more general? It doesn't seem like we need mutability here, so we could probably do `AsRef<Handle<A>>`.

Reply:
Could you explain how I could implement that?

Reply:
```suggestion
impl<T: AsRef<Handle<A>>, A: Asset> From<T> for AssetId<A> {
```
Does this work?

Reply:
```suggestion
impl<T: Borrow<Handle<A>>, A: Asset> From<T> for AssetId<A> {
```

It appears that `Borrow` is preferred: https://www.reddit.com/r/rust/comments/xqrz14/asref_vs_borrow/

Reply:
Commited!